flow/controllers/imitation_learning/imitating_network.py

```diff
@@ -0,0 +1,102 @@
+import numpy as np
+import tensorflow as tf
+from utils import *
+import tensorflow_probability as tfp
+from flow.controllers.base_controller import BaseController
+from replay_buffer import ReplayBuffer
+
+
+class ImitatingNetwork():
+    """
+    Neural network which learns to imitate another given expert controller.
+    """
+    # Implementation in Tensorflow
+
+    def __init__(self, sess, action_dim, obs_dim, num_layers, size, learning_rate, replay_buffer_size, training = True, inject_noise=0, noise_variance=0.5, policy_scope='policy_vars'):
+
+        self.sess = sess
+        self.action_dim = action_dim
+        self.obs_dim = obs_dim
+        self.num_layers = num_layers
+        self.size = size
+        self.learning_rate = learning_rate
+        self.training = training
+        self.inject_noise=inject_noise
+        self.noise_variance = noise_variance
+
+        with tf.variable_scope(policy_scope, reuse=tf.AUTO_REUSE):
+            self.build_network()
+
+        if self.training:
+            self.replay_buffer = ReplayBuffer(replay_buffer_size)
+        else:
+            self.replay_buffer = None
+
+        self.policy_vars = [v for v in tf.all_variables() if policy_scope in v.name and 'train' not in v.name]
+        self.saver = tf.train.Saver(self.policy_vars, max_to_keep=None)
+
+    def build_network(self):
+        """
+        Defines neural network for choosing actions.
+        """
+        self.define_placeholders()
+        self.define_forward_pass()
+        if self.training:
+            with tf.variable_scope('train', reuse=tf.AUTO_REUSE):
+                self.define_train_op()
+
+
+    def define_placeholders(self):
+        """
+        Defines input, output, and training placeholders for neural net
+        """
+        self.obs_placeholder = tf.placeholder(shape=[None, self.obs_dim], name="obs", dtype=tf.float32)
+        self.action_placeholder = tf.placeholder(shape=[None, self.action_dim], name="action", dtype=tf.float32)
```

Comment:
So for stochastic algorithms, they are parametrized by a mean and standard deviation of a gaussian that you sample from. It'd be cool to add this as an option here so we can use PPO

Reply:
This current implementation can be used for deterministic algorithms like DDPG and TD3 which is great